core/build.rs

```diff
@@ -45,10 +66,47 @@ fn build_packages() {
             let exit = f.wait().expect("failed to launch wasm build");
             if !exit.success() {
                 println!("cargo:warning=failed to build package: {name}")
+            } else {
+                #[cfg(all(
+                    feature = "bundle_std_packages",
+                    feature = "native",
+                    feature = "precompile_wasm"
+                ))]
+                precompile_wasm(&name, &out_path, &engine);
             }
         });
 }
 
+#[cfg(all(
+    feature = "bundle_std_packages",
+    feature = "native",
+    feature = "precompile_wasm"
+))]
+/// This function pre-compiles a wasm file for a package with the given name, and makes a new file
+/// `{name}-precompiled.wasm` which holds the serialized data for the precompiled package.
+fn precompile_wasm(name: &str, output_path: &Path, engine: &Engine) {
+    let in_path = output_path
+        .join(name)
+        .join("wasm32-wasi")
+        .join("release")
+        .join(name.to_string() + ".wasm");
```

Comment:
Might be cheaper/cleaner to do `&format!("{name}.wasm")` but does not really matter. Or perhaps you can push and then use set_extension to the pathbuf.

Reply:
Yeah, I can do format strings, sure! That's what I usually do, however I thought that since this only was one extra call to .to_string(), it wouldn't make that much of a difference. But I fixed it here and at the other place